### DISCUSSION TITLE
Release v0.1.17b6

### DISCUSSION THREAD

**Discussion by ehendrix23** [ehendrix23/tesla_dashcam#162] [Announcements]
- Added debug information to show what platform and processor were identified, can help in troubleshooting.
- When system information returned is MacOS Intel then check actual processor with sysctl to ensure it is Intel and not Apple Silicon
- When running the binary on Apple Silicon provide a message to download an Apple Silicon ffmpeg instead and use --ffmpeg parameter.

Unless something comes up this will be turned into final release v0.1.17

<hr /><em>This discussion was created from the release <a href='https://github.com/ehendrix23/tesla_dashcam/releases/tag/v0.1.17b6'>Release v0.1.17b6</a>.</em>

**Comment by rmyadsk** [comment]
Oh, not good. 

> myadsk@Ryans-Mac-Mini Downloads % mv tesla_dashcam /Applications      
rmyadsk@Ryans-Mac-Mini Downloads % chmod +x /Applications/tesla_dashcam
rmyadsk@Ryans-Mac-Mini Downloads % /Applications/tesla_dashcam -h
zsh: killed     /Applications/tesla_dashcam -h
rmyadsk@Ryans-Mac-Mini Downloads %

**Reply by ehendrix23** [reply]
Hmm, it is running fine on mine. Only thing really I added that I can see causing this is running of sysctl.

I just created a new "temp" binary in the hopes to capture a bit more information in at least it printing out when it is running sysctl and when completed to determine if that is where it fails on.
It is located within the DEV branch under bundles/MacOS/tesla_dashcam. Download that and let me know what you see.
Further, also before trying open Console so can see when you run it if anything is shown in there. Be it in Console itself or within system.log. Somewhere it would be reporting some more information that might be helpful.

**Reply by rmyadsk** [reply]
Nothing obvious in the console, and initially this temp binary resulted in the same "killed" result when run from a terminal. Instead I found after I fired it once from double-clicking its location (/Applications on my system) and assuring MacOS I wanted to open it, it seems to run okay (even from the terminal). So I wonder if some weird MacOS syspolicy was interfering?

From running with LOGLEVEL=DEBUG, I see:

2021-08-15 15:12:15,326:DEBUG:	__main__	Platform is darwin
2021-08-15 15:12:15,326:DEBUG:	__main__	Processor is arm

Watching "top" looked like ffmpeg using ~750% CPU, which suggests hardware acceleration is rightfully disabled. Product video appears well-formed, and no invocation of--no-gpu or -ffmpeg is necessary.

To check, I ran the same clip again again invoking --ffmpeg (specifying the OSXExperts ffmpeg) and the resulting file is 562.5MB vice 667.4MB without specifying ffmpeg (nine-minute clip). Quality appears the same.

**Comment by rmyadsk** [comment]
Okay, back with the non-temp binary. Some testing and I think I have it worked out. First, run the executable by right-clicking on its location in Finder and selecting "Open", then accept the warning from MacOS. This will result in a terminal window:

> /Applications/tesla_dashcam ; exit;
> rmyadsk@Ryans-Mac-Mini ~ % /Applications/tesla_dashcam ; exit;
> Internal ffmpeg version is used which has been compiled for Intel Macs. Better results in both performance and size can be achieved by downloading an Apple Silicon compiled ffmpeg from: https://www.osxexperts.net and providing it leveraging the --ffmpeg parameter.
> Monitoring for TeslaCam Drive to be inserted. Press CTRL-C to stop

From there, that terminal can be killed, and the executable can be run from a terminal (or, say, a shell script passing it parameters). 

This quirk may be specific to macOS Monterey (I'm running 12.0 Beta 5 - build 21A5304g) or something else specific to my system.

Some results with the previous clip above, using the non-temp binary:

Specifying OSExperts ffmpeg: 4 minutes 6 seconds rendering time, 562.5MB product
Not specifying ffmpeg: 6 minutes 42 seconds rendering time, 667.4MB product

So agreed, running v0.1.17b6 specifying the OSExperts ffmpeg (as the non-temp binary suggests) is the best result for users of M1 silicon.

**Comment by rmyadsk** [comment]
One thing I have noticed with 0.1.17b6: it seems to always throw the "Internal ffmpeg version is used" error, even if it is not. This while invoking --ffmpeg and pointing to the OSXExperts.net static build:

> rmyadsk@Ryans-Mac-Mini Samsung % ./runblue.sh 
> Starting transcoding at Mon Aug 16 23:35:14 EDT 2021
> 2021-08-16 23:35:18,217:DEBUG:	__main__	Arguments : Namespace(source=['/Volumes/Samsung/blue'], loglevel='DEBUG', temp_dir=None, system_notification=True, display_ts=False, skip_existing=False, delete_source=False, exclude_subdirs=False, monitor=False, monitor_once=False, monitor_trigger=None, layout='DIAMOND', perspective=False, clip_scale=None, rear_or_mirror=None, swap_leftright=None, swap_frontrear=False, background='black', title_screen_map=True, no_front=False, no_left=False, no_right=False, no_rear=False, no_timestamp=False, halign='RIGHT', valign='BOTTOM', font='/Library/Fonts/Arial Unicode.ttf', fontsize=None, fontcolor='white', text_overlay_fmt='{local_timestamp_rolling}\\nCity: {event_city}\\nTrigger: {event_reason}', timestamp_format='%x %X', start_timestamp=None, end_timestamp=None, start_offset=None, end_offset=None, sentry_offset=False, output='/Volumes/Samsung/blue/product', motion_only=False, chapter_offset=0, merge_timestamp_format='%Y-%m-%d_%H_%M', keep_intermediate=False, keep_events=False, set_moviefile_timestamp='START', faststart=False, quality='MEDIUM', compression='medium', fps=24, ffmpeg='/Applications/ffmpeg', check_for_updates=False, no_check_for_updates=False, include_beta=False)
> 2021-08-16 23:35:18,217:DEBUG:	__main__	Platform is darwin
> 2021-08-16 23:35:18,217:DEBUG:	__main__	Processor is arm
> 2021-08-16 23:35:18,282:DEBUG:	urllib3.connectionpool	Starting new HTTPS connection (1): api.github.com:443
> 2021-08-16 23:35:18,395:DEBUG:	urllib3.connectionpool	https://api.github.com:443 "GET /repos/ehendrix23/tesla_dashcam/releases/latest HTTP/1.1" 200 1781
> Internal ffmpeg version is used which has been compiled for Intel Macs. Better results in both performance and size can be achieved by downloading an Apple Silicon compiled ffmpeg from: https://www.osxexperts.net and providing it leveraging the --ffmpeg parameter.
> 2021-08-16 23:35:18,402:DEBUG:	__main__	Video Settings {'source_folder': ['/Volumes/Samsung/blue'], 'exclude_subdirs': False, 'output': '/Volumes/Samsung/blue/product', 'target_folder': '/Volumes/Samsung/blue/product', 'target_filename': None, 'temp_dir': None, 'run_type': 'RUN', 'merge_subdirs': False, 'merge_group_template': None, 'merge_timestamp_format': '%Y-%m-%d_%H_%M', 'chapter_offset': 0, 'movie_filename': None, 'set_moviefile_timestamp': 'START', 'keep_intermediate': False, 'keep_events': False, 'notification': True, 'movie_layout': 'DIAMOND', 'movie_speed': '', 'video_encoding': ['-c:v', 'libx264'], 'movie_encoding': 'x264', 'fps': 24, 'movie_compression': 'medium', 'movie_quality': 'MEDIUM', 'background': ';color=duration={duration}:s={width}x{height}:c=black, fps=24 ', 'ffmpeg_exec': '/Applications/ffmpeg', 'base': 'color=duration={duration}:s=1920x984:c=black, fps=24 [base]', 'video_layout': <__main__.Diamond object at 0x10fc5ad00>, 'clip_positions': ';[base][left] overlay=eof_action=pass:repeatlast=0:x=0:y=252 [left1];[left1][front] overlay=eof_action=pass:repeatlast=0:x=640:y=0 [front1];[front1][right] overlay=eof_action=pass:repeatlast=0:x=1280:y=252 [right1];[right1][rear] overlay=eof_action=pass:repeatlast=0:x=640:y=504 [rear1]', 'ffmpeg_text_overlay': ";[rear1] drawtext=fontfile=/Library/Fonts/Arial Unicode.ttf:fontcolor=white:fontsize=24:borderw=2:bordercolor=black@1.0:x=(w-text_w):y=(h-(text_h)-10):text='__USERTEXT__' [tmp0]", 'text_overlay_format': '{local_timestamp_rolling}\\nCity: {event_city}\\nTrigger: {event_reason}', 'timestamp_format': '%x %X', 'ffmpeg_speed': '', 'ffmpeg_motiononly': '', 'movflags_faststart': True, 'input_clip': 'tmp0', 'other_params': ['-preset', 'medium', '-crf', '20', '-c:v', 'libx264'], 'left_camera': 'setpts=PTS-STARTPTS, scale=640x480 , hflip [left]', 'front_camera': 'setpts=PTS-STARTPTS, scale=640x480  [front]', 'right_camera': 'setpts=PTS-STARTPTS, scale=640x480 , hflip [right]', 'rear_camera': 'setpts=PTS-STARTPTS, scale=640x480 , hflip [rear]', 'start_timestamp': None, 'start_offset': None, 'end_timestamp': None, 'end_offset': None, 'sentry_offset': False, 'skip_existing': False} 

The video does appear to be of the smallest file and shortest processing time, though.

**Reply by ehendrix23** [reply]
Thx, fixed it for release.
